Automattic/jetpack
# My Jetpack product interstitial: Don't show intro offer price if user is not eligible for offer

This PR updates the My Jetpack product interstitials so that if the user is not eligible for the intro offer, then the intro offer price is not shown.

Prior to this PR, intro offer prices were shown, regardless whether the user was eligible to receive the offer or not.

### Screenshot:

![Screen Shot 2024-09-16 at 12 03 59](https://github.com/user-attachments/assets/6f7c9ab9-52ac-49dc-a5e1-4405209f0eac)






Fixes: https://github.com/Automattic/jetpack-marketing/issues/989

## Proposed changes:

* Don't show intro offer price if user is not eligible for offer.
* Show a small note: "* Not eligible for introductory discount" under the price

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






Prerequisite: Patch your WPCOM sandbox with D161504-code, **only if it's not already shipped**.

1. Checkout this branch via the Jetpack Beta plugin or your local dev environment
2. Connect Jetpack (site and user)
3. Go to `/wp-admin/admin.php?page=my-jetpack#/add-backup`
4. See that the intro-offer price is showing (for both Backup and Security).
![image](https://github.com/user-attachments/assets/8dca62f2-b192-4f66-b44b-c5cfe3347e7d)
5. Now purchase Backup for the intro offer price (using credits is fine).
6. Go to Store Admin and now remove the Backup subscription you just purchased for that site.
7. SSH into your testing site and run the following command `wp user meta delete {username/user ID} "my-jetpack-cache"` (This cache will normally be refreshed once per day, which I think is enough in this specific scenario)
NOTE: If you are using a Jurassic ninja site, the username will be `demo`
8. Now reload My Jetpack and click "Get plan" on the Backup card to open the product interstitial again.
9. See and verify that the intro offer (for Backup) is no longer showing, but instead it's the full un-discounted price.
![image](https://github.com/user-attachments/assets/c26f0daa-7d56-4069-95d6-742d08e7bc71)
10. Go to `/wp-admin/admin.php?page=my-jetpack#/add-protect`
11. Repeat steps 4-9 to test the intro discount with a Product interstitial table as well
![image](https://github.com/user-attachments/assets/821718e4-4b38-495f-81bc-9191aea774f4)
![image](https://github.com/user-attachments/assets/cc26fdfd-be93-4839-81fc-043693e56a7c)
